src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -101,14 +101,19 @@
      */
     private /*almost final*/ transient CpsFlowExecution execution;
 
+    /**
+     * Persistent version of {@link #runtimeThreads}.
+     */
+    private Map<Integer, CpsThread> threads;
```

Comment:
and correspondingly

```suggestion
    private volatile Map<Integer, CpsThread> threads;
```

just in case (though I think `writeReplace` will be called from the CPS VM thread only)